source/_guides/6-php-client-intro.md

```diff
@@ -0,0 +1,207 @@
+      "name": "resize",
+      "options": {
+        "width": 200
+      }
+    }
+  ]
+}' | rokka stacks create $YOUR_ORG test-stack
+``` 
+
+And then you can render an image simply with using that stack name instead of `dynamic` from above: `https://$YOUR_ORG.rokka.io/test-stack/dba893.jpg`
+
+If you want to overwrite some operation parameters or stack options, you can do that directly in the URL, eg.
+
+`https://$YOUR_ORG.rokka.io/test-stack/resize-width-300--options-dpr-2/dba893.jpg` would return an image with a width of 300 px but in a device-pixel-ratio of 2 (for retina), which makes the picture actually 600px wide. This way you can easily have complex stacks (cropping and resizing for example) and just overwrite some parameters via URL, if you need to.
+
+## The PHP library
+
+After having uploaded a first image and created a first stack, let's finally move to the PHP part.
+
+
+### Installation
+
+Install the library using [composer](https://getcomposer.org/):
+
+```
+composer require rokka/client
+```
+
+### First steps
+
+You may use rokka within a bigger framework (like Symfony), a CMS (like WordPress or Drupal) or within Twig, the following steps may not be needed or different, but they give a good feeling, how it works.
+
+The easiest way to let the library generate the needed URLs is by using the `\Rokka\Client\TemplateHelper` class.
+
+```
+<?php
+require_once("vendor/autoload.php");
+use Rokka\Client\TemplateHelper;
+$rokka = new TemplateHelper('$YOUR_ORG', '$YOUR_API_KEY');
+?>
+<html><body>
+
+<img src="<?= $rokka->getStackUrl('dba893', 'test-stack', 'jpg');?>"> <br/>
+
+</body>
+</html>
+```
+
+Alternative, you can also just use the `\Rokka\Client\UriHelper:composeUri ` static method.
+
+```
+<img src="https://liip-development.rokka.io<?= UriHelper::composeUri(['stack' => 'test-stack', 'hash' => 'dba893', 'format' => 'jpg']);?>">
+```
+
+### Working with "local" images
+
+Uploading images manually to rokka is not a very efficient way to deliver them.  You could integrate the uploading part into your Framework/CMS and use the PHP API for that (see below about that). 
+
+Or you can use the built in functionialities of the  `TemplateHelper` class to help you in that. Some methods of that class not only take a rokka image hash as first argument, but also a path to an existing local file, a `SplFileInfo`  object or an object which implements `\Rokka\Client\LocalImage\LocalImageAbstract` (more about that later)
+
+The TemplateHelper then automatically uploads the image to rokka, if it can't find a corresponding hash locally and does all the rest for you. 
+```
+<img src="<?= $rokka->getStackUrl('images/foo.jpg', 'test-stack', 'jpg');?>"> 
+```
+
+#### Store hashes somewhere else with TemplateHelperCallbacks
+
+By default it stores the hash in a text file next to the local image, but that's maybe not what you need. You can write your own class implementing `\Rokka\Client\TemplateHelperCallbacksAbstract` to change that behaviour.
```

Comment:
is that new? that's nice :)

Reply:
In development phase since ages, but finally planning to publish it.

Reply:
@chregu the class name changed IIRC :)